src/org/intellij/erlang/utils/ErlangModulesUtil.java

```diff
@@ -117,6 +132,27 @@ private static ErlangModule doGetErlangModule(@NotNull Project project,
     return erlangModules;
   }
 
+  private static Collection<VirtualFile> addFiles(@NotNull Module module,
+                                                  boolean onlyTest,
+                                                  @NotNull final Collection<VirtualFile> files,
+                                                  final ErlangFileType type) {
+    ModuleRootManager rootManager = ModuleRootManager.getInstance(module);
+    ModuleFileIndex moduleFileIndex = rootManager.getFileIndex();
+    Convertor<VirtualFile, Boolean> sourceDirectoriesFilter = onlyTest ? getTestDirectoriesFilter(moduleFileIndex) : getSourceDirectoriesFilter(moduleFileIndex);
+    Processor<VirtualFile> modulesCollector = new Processor<VirtualFile>() {
+      @Override
+      public boolean process(VirtualFile virtualFile) {
+        if (virtualFile.getFileType() == type) {
+          files.add(virtualFile);
+        }
+        return true;
+      }
+    };
+    for (VirtualFile sourceRoot : rootManager.getSourceRoots(onlyTest)) {
+      VfsUtilCore.processFilesRecursively(sourceRoot, modulesCollector, sourceDirectoriesFilter);
+    }
```

Comment:
Well, you copied this code from ```org.intellij.erlang.utils.ErlangModulesUtil#addErlangFiles``` and simply a different collector. Why haven't you extracted common logic to a separate method?